### PR TITLE
Icepack install tweaks

### PR DESCRIPTION
**Pull request by dham**
Firedrake provides inbuilt support for installing icepack. This means Firedrake and icepack can be installed in a single line, which makes the icepack install instructions shorter.

As an aside, I note that the icepack install instructions also install umfpack. Firedrake builds mumps into PETSc to provide a good parallel direct solver. Is there something about umfpack which means we need that in addition to mumps?

### CODE REVIEW

File: doc/source/installation.rst [hunk 1]
```diff
@@ -9,20 +9,15 @@ Quick start
...
 
-* Clone and install icepack::
-
-   git clone https://github.com/icepack/icepack
-   cd icepack
-   pip3 install -e .
-
 * Run the icepack unit tests to make sure everything's working::
 
+   cd firedrake/src/icepack
```
Comment (danshapero) [inline doc/source/installation.rst]:
```suggestion
   cd <path/to/firedrake>/src/icepack
```

will make the instructions more consistent with what I have above about activating the virtual environment.

**Comment by danshapero** [comment]
Thanks for the PR! I remember getting errors when solving systems with `pc_type='lu'` and telling PETSc to install umfpack would fix it. I'll try a build without umfpack and change the instructions if it works.

I already have the contents of the requirements file that you added in the `install_requires` field in the setup script -- is there a reason to have that information repeated? Should it be removed from the setup script then? Every time I look into this I get conflicting information and give up.

**Comment by dham** [comment]
> Thanks for the PR! I remember getting errors when solving systems with `pc_type='lu'` and telling PETSc to install umfpack would fix it. I'll try a build without umfpack and change the instructions if it works.

Just lu will give you PETSc's native LU, which may or may not be good enough. To use MUMPS, you add:
```
"pc_factor_mat_solver_type": "mumps"
```

> 
> I already have the contents of the requirements file that you added in the `install_requires` field in the setup script -- is there a reason to have that information repeated? Should it be removed from the setup script then? Every time I look into this I get conflicting information and give up.

`install_requires` is what pip uses, so you do want that. Firedrake-install does its own dependency management (for some very weak definition of "management") which depends on requirements files. For current purposes I think that you need both, which I agree is an unfortunate duplication.

File: doc/source/installation.rst (hunk 1 above)
Comment (dham) [reply doc/source/installation.rst]: I just pushed a slightly different fix. If one uses `$VIRTUAL_ENV` then the user can just paste the code.

**Pull request merged**